language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/workspace/BallerinaWorkspaceManager.java

```diff
@@ -536,6 +554,109 @@ public String uriScheme() {
         return "file";
     }
 
+    @Override
+    public Optional<Process> run(Path filePath) throws IOException {
+        Optional<ProjectPair> projectPairOpt = projectPair(projectRoot(filePath));
+        if (projectPairOpt.isEmpty()) {
+            clientLogger.logWarning("Run command execution aborted because project is not loaded");
```

Comment:
Shouldn't this be `logError()`

Reply:
To call log error, I need an exception.

Reply:
Shall we use the following or introduce something similar. https://github.com/ballerina-platform/ballerina-lang/blob/master/language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/exception/UserErrorException.java

Reply:
Ok